File: the-efs-lambda/csharp/src/TheEfsLambda/TheEfsLambdaStack.cs
using Amazon.CDK;
using Lambda = Amazon.CDK.AWS.Lambda;
using APIGv2 = Amazon.CDK.AWS.APIGatewayv2;
using EC2 = Amazon.CDK.AWS.EC2;
using EFS = Amazon.CDK.AWS.EFS;
using APIGv2Integration = Amazon.CDK.AWS.APIGatewayv2.Integrations;

namespace TheEfsLambda
{
    public class TheEfsLambdaStack : Stack
    {
        // declaring all constructors
        readonly private EC2.Vpc _vpc;
        readonly private EFS.FileSystem _fileSystem;
        readonly private Lambda.Function _functionProxyHandler;
        readonly private APIGv2.HttpApi _apiGateway;

        internal TheEfsLambdaStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
        {
            // EFS needs to be setup in a VPC with 2Azs
            _vpc = new EC2.Vpc(this, "Vpc", new EC2.VpcProps
            {
                MaxAzs = 2
            });

            // Create a file system in EFS to store information
            _fileSystem = new EFS.FileSystem(this, "Filesystem", new EFS.FileSystemProps
            {
                Vpc = _vpc,
                RemovalPolicy = RemovalPolicy.DESTROY
            });

            // Create a access point to EFS
            EFS.AccessPoint accessPoint;
            accessPoint = _fileSystem.AddAccessPoint("AccessPoint", new EFS.AccessPointOptions
            {
                CreateAcl = new EFS.Acl { OwnerGid = "1001", OwnerUid = "1001", Permissions = "750"},
                Path = "/export/lambda",
                PosixUser = new EFS.PosixUser { Gid = "1001", Uid = "1001", }
            });

            // Create the lambda function
            _functionProxyHandler = new Lambda.Function(this, "efsLambdaFunction", new Lambda.FunctionProps
            {
                Runtime = Lambda.Runtime.PYTHON_3_8,
                Code = Lambda.Code.FromAsset("lambda_fns"),
                Handler = "message_wall.lambda_handler",
                Vpc = _vpc,
                Filesystem = Lambda.FileSystem.FromEfsAccessPoint(accessPoint, "/mnt/msg")
            });

            // Api Gateway HTTP integration
            _apiGateway = new APIGv2.HttpApi(this, "EFS Lambda", new APIGv2.HttpApiProps
            {
                DefaultIntegration = new APIGv2Integration.LambdaProxyIntegration(new APIGv2Integration.LambdaProxyIntegrationProps
                {
                    Handler = _functionProxyHandler
                })
            });


            // Output to CFN
            new CfnOutput(this, "HTTP API Url", new CfnOutputProps
            {
                Value = _apiGateway.Url
            });
        }
    }
}

File: the-efs-lambda/java/src/main/java/com/cdkpatterns/TheEfsLambdaStack.java
package com.cdkpatterns;

import software.amazon.awscdk.core.*;
import software.amazon.awscdk.services.lambda.Code;
import software.amazon.awscdk.services.lambda.Function;
import software.amazon.awscdk.services.lambda.Runtime;
import software.amazon.awscdk.services.ec2.Vpc;
import software.amazon.awscdk.services.efs.FileSystem;
import software.amazon.awscdk.services.efs.AccessPoint;
import software.amazon.awscdk.services.efs.AccessPointOptions;
import software.amazon.awscdk.services.efs.Acl;
import software.amazon.awscdk.services.efs.PosixUser;
import software.amazon.awscdk.services.apigatewayv2.HttpApi;
import software.amazon.awscdk.services.apigatewayv2.integrations.LambdaProxyIntegration;

public class TheEfsLambdaStack extends Stack {
    public TheEfsLambdaStack(final Construct scope, final String id) {
        this(scope, id, null);
    }

    public TheEfsLambdaStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        // EFS needs to be setup in a VPC
        Vpc vpc = Vpc.Builder.create(this, "Vpc")
        		.maxAzs(2)
        		.build();
        
        // Create a file system in EFS to store information
        FileSystem fileSystem = FileSystem.Builder.create(this, "FileSystem")
        		.vpc(vpc)
        		.removalPolicy(RemovalPolicy.DESTROY)
        		.build();
        
        // Create a access point to EFS
        AccessPoint accessPoint = fileSystem.addAccessPoint("AccessPoint", 
        		AccessPointOptions.builder()
        		.createAcl(
        				Acl
        				.builder()
        				.ownerGid("1001").ownerUid("1001").permissions("750")
        				.build())
        		.path("/export/lambda")
        		.posixUser(
        				PosixUser
        				.builder()
        				.gid("1001").uid("1001")
        				.build())
        		.build());
        
        // Create the lambda function
        Function efsLambda = Function.Builder.create(this, "efsLambdaFunction")
        		.runtime(Runtime.PYTHON_3_8)
        		.code(Code.fromAsset("lambda_fns"))
        		.handler("message_wall.lambda_handler")
        		.vpc(vpc)
        		.filesystem(software.amazon.awscdk.services.lambda.FileSystem.fromEfsAccessPoint(accessPoint, "/mnt/msg"))
        		.build();

        // Api Gateway HTTP integration
        HttpApi httpApi = HttpApi.Builder.create(this, "EFS Lambda")
        		.defaultIntegration(
        				LambdaProxyIntegration
        				.Builder
        				.create()
        				.handler(efsLambda)
        				.build())
        		.build();
        
        CfnOutput.Builder.create(this, "HTTP API Url")
        .value(httpApi.getUrl())
        .build();
    }
}
